### PR TITLE
fix: Setting window fails to appear front

### PR DESCRIPTION
**Pull request by phlpsong**
#382 

Introduce one new package: https://github.com/orchetect/SettingsAccess

This workaround does nothing if running from Xcode, but running as a standalone app it will work according to doc.

But it seems to work well both from Xcode build or archive based on my test.

### CODE REVIEW

**Comment by CanglongCl** [comment]
Did it fix in macOS 13.0? It seems only affect macOS 14.0+.

**Comment by phlpsong** [comment]
> Is it fixed in macOS 13.0? It seems only affect macOS 14.0+.

Yes, I thought it only affects on macOS 14.0+ using `SettingLink`. 

I think `NSApp.activate(ignoringOtherApps: true)` should work on macOS 13 too, could you have a try if you're using macOS 13?

**Comment by CanglongCl** [comment]
> > Is it fixed in macOS 13.0? It seems only affect macOS 14.0+.
> 
> Yes, I thought it only affects on macOS 14.0+ using `SettingLink`.
> 
> I think `NSApp.activate(ignoringOtherApps: true)` should work on macOS 13 too, could you have a try if you're using macOS 13?

I do not use macOS 13 currently. Any other one can help?

**Comment by Jerry23011** [comment]
I can setup a virtual machine later

I'll check the release mode only since it's too complicated to set up Xcode from gound up.

**Comment by Jerry23011** [comment]
The issue still exists on macOS 13

I tested it on a macOS 13.6 VM

https://github.com/tisfeng/Easydict/assets/89069957/f5bda160-26f5-47b3-afa0-c5614e970748

The workaround is working on a macOS 14.4 beta 1 VM, I can see there's a workaround going, but it's pretty stable.

https://github.com/tisfeng/Easydict/assets/89069957/4bcfd8dc-c5f1-4faa-aa62-c54948d7e47c

**Comment by phlpsong** [comment]
> The issue still exists on macOS 13
> 
> I tested it on a macOS 13.6 VM
> 
>  macOS.13.6.mov 
> The workaround is working on a macOS 14.4 beta 1 VM, I can see there's a workaround going, but it's pretty stable.
> 
>  macOS.14.4.beta.1.mov

I just push a new commit to apply `NSApp.activate(ignoringOtherApps: true)` below macOS 14. 

Could u give it a try?

**Comment by Jerry23011** [comment]
Yep! Now it works on 13.6

**Review by Jerry23011 — DISMISSED** [review]

**Comment by phlpsong** [comment]
> Yep! Now it works on 13.6

Much appreciated!

**Review by CanglongCl — DISMISSED** [review]

**Comment by tisfeng** [comment]
I tested this code, it works.

So, is this a bug in the use of SettingsLink in SwiftUI? 🤔

File: Easydict/NewApp/View/MenuItemView.swift [hunk 1]
```diff
@@ -92,10 +94,18 @@ struct MenuItemView: View {
...
+            } preAction: {
+                NSLog("打开设置")
+                NSApp.activate(ignoringOtherApps: true)
+            } postAction: {
+                // nothing to do
+            }
         } else {
             Button("Settings...") {
                 NSLog("打开设置")
+                NSApp.activate(ignoringOtherApps: true)
```
Comment (tisfeng) [inline Easydict/NewApp/View/MenuItemView.swift:108]: `activate(ignoringOtherApps:)` will be deprecated, can we use NSApp.activate instead?

https://developer.apple.com/documentation/appkit/nsapplication/1428468-activate

```swift
    /// Makes the receiver the active app.
    @available(macOS, introduced: 10.0, deprecated: 100000, message: "This method will be deprecated in a future release. Use NSApp.activate instead.")
    open func activate(ignoringOtherApps flag: Bool)
```

**Review by tisfeng — CHANGES_REQUESTED** [review]

File: Easydict/NewApp/View/MenuItemView.swift (hunk 1 above)
Comment (phlpsong) [reply Easydict/NewApp/View/MenuItemView.swift:108]: `NSApp.active()` is only available in macOS 14.

I tested in macOS14, and it does not work well both from Xcode build and release.

**Comment by phlpsong** [comment]
> I tested this code, it works.
> 
> So, is this a bug in the use of SettingsLink in SwiftUI? 🤔

It can be said that Apple does not provide extra interfaces for `SettingsLink` to let us control the activation of window.

File: Easydict/NewApp/View/MenuItemView.swift (hunk 1 above)
Comment (tisfeng) [reply Easydict/NewApp/View/MenuItemView.swift:108]: Ok, let's keep it.

**Comment by tisfeng** [comment]
Oh, I see.

**Review by tisfeng — APPROVED** [review]

File: Easydict/NewApp/View/MenuItemView.swift [hunk 2]
```diff
@@ -24,6 +25,7 @@ final class MenuItemStore: ObservableObject {
 @available(macOS 13, *)
 struct MenuItemView: View {
     @ObservedObject private var store = MenuItemStore()
+    @Environment(\.openSettings) private var openSettings
```
Comment (tisfeng) [inline Easydict/NewApp/View/MenuItemView.swift:28]: It seems that this variable is not used?

**Review by tisfeng — CHANGES_REQUESTED** [review]

File: Easydict/NewApp/View/MenuItemView.swift (hunk 2 above)
Comment (phlpsong) [reply Easydict/NewApp/View/MenuItemView.swift]: Yes, forgot to remove it.  Another workaround used this at first.

**Review by AkaShark — APPROVED** [review]

**Review by Jerry23011 — APPROVED** [review]

**Review by tisfeng — APPROVED** [review]

**Pull request merged**